tests/python/openassetio/test_traits.py

```diff
@@ -1,231 +0,0 @@
-        a_specification.setTraitProperty("second_trait", "a bool", True)
-
-        assert a_specification.getTraitProperty("first_trait", "a string") == "string"
-        assert isinstance(a_specification.getTraitProperty("first_trait", "a string"), str)
-        assert a_specification.getTraitProperty("second_trait", "an int") == 1
-        assert isinstance(a_specification.getTraitProperty("second_trait", "an int"), int)
-        assert a_specification.getTraitProperty("first_trait", "a float") == 1.0
-        assert isinstance(a_specification.getTraitProperty("first_trait", "a float"), float)
-        assert a_specification.getTraitProperty("second_trait", "a bool") is True
-
-    def test_when_key_is_not_found_then_get_returns_None(self, a_specification):
-        assert a_specification.getTraitProperty("first_trait", "a string") is None
-
-    def test_when_trait_is_not_found_then_IndexError_raised(self, a_specification):
-        with pytest.raises(IndexError):
-            a_specification.setTraitProperty("unknown_trait", "a string", "string")
-
-        with pytest.raises(IndexError):
-            _ = a_specification.getTraitProperty("unknown_trait", "a string")
-
-    def test_when_value_is_not_supported_then_set_raises_TypeError(self, a_specification):
-        with pytest.raises(TypeError):
-            a_specification.setTraitProperty("first_trait", "unknown type", object())
-
-        with pytest.raises(TypeError):
-            a_specification.setTraitProperty("first_trait", "unknown type", None)
-
-
-class Test_Specification_equality:
-    def test_when_comparing_with_same_data_then_are_equal(self):
-        spec_a = Specification({"a_trait"})
-        spec_a.setTraitProperty("a_trait", "a_property", 1)
-        spec_b = Specification({"a_trait"})
-        spec_b.setTraitProperty("a_trait", "a_property", 1)
-        assert spec_a == spec_b
-
-    def test_when_comparing_with_different_property_value_then_are_not_equal(self):
-        spec_a = Specification({"a_trait"})
-        spec_a.setTraitProperty("a_trait", "a_property", 1)
-        spec_b = Specification({"a_trait"})
-        spec_b.setTraitProperty("a_trait", "a_property", 2)
-        assert spec_a != spec_b
-
-    def test_when_comparing_with_subset_of_properties_then_are_not_equal(self):
-        spec_a = Specification({"a_trait"})
-        spec_a.setTraitProperty("a_trait", "a_property", 1)
-        spec_b = Specification({"a_trait"})
-        spec_b.setTraitProperty("a_trait", "a_property", 1)
-        spec_b.setTraitProperty("a_trait", "another_property", 1)
-        assert spec_a != spec_b
-
-    def test_when_comparing_with_different_trait_then_are_not_equal(self):
-        spec_a = Specification({"a_trait"})
-        spec_b = Specification({"another_trait"})
-        assert spec_a != spec_b
-        assert spec_a != spec_b
-
-
-def test_BlobTrait_traitId():
-    assert trait.BlobTrait.kId == "blob"
-
-
-class Test_BlobTrait_isValid:
```

Comment:
[plan] Removing `BlobTrait` tests means that `TraitBase` is now untested. 

We had avoided testing `TraitBase` in isolation because that would mean C++-specific tests, rather than testing through the Python interface.  

We now have most of the boilerplate required for testing C++ specifics (i.e. copy-paste the C API testing boilerplate), so we could add `TraitBase`-specific tests without too much fuss.

Alternatively, we can leave this until we start developing traits in C++... its conceivable `TraitBase` would be redesigned or disappear, and if it's not being used maybe there's no point investing the time to test it.

Reply:
Are you happy to push this back to when we work on our first `c++` trait, incase, as you say, `TraitBase` changes in any way?

Reply:
Added [this note](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/358/commits/ff24bb7adf96b6a92ecb3242f50f2623ce4dc716).

Reply:
That's fine, thanks.